test/test_operation.cpp

```diff
@@ -135,3 +138,106 @@ TEST(Operation, IsDiagonalGate) {
   const qc::StandardOperation op2(0, qc::Z);
   EXPECT_TRUE(op2.isDiagonalGate());
 }
+
+TEST(StandardOperation, Move) {
+  const qc::StandardOperation moveOp({0, 1}, qc::OpType::Move);
+  EXPECT_EQ(moveOp.getTargets().size(), 2);
+  EXPECT_EQ(moveOp.getNqubits(), 2);
+}
+
+TEST(AodOperation, Activate) {
+  // activate at position 0, dimension X, start 0.0, end 1.0
+  const qc::AodOperation activate(qc::OpType::AodActivate, {0},
+                                  {qc::Dimension::X}, {0.0}, {1.0});
+  EXPECT_EQ(activate.getNqubits(), 1);
+  EXPECT_EQ(activate.getStarts(qc::Dimension::X).at(0), 0.0);
+  EXPECT_EQ(activate.getEnds(qc::Dimension::X).at(0), 1.0);
+}
+
+TEST(AodOperation, Deactivate) {
+  // deactivate at position 0,1 dimension Y, start 0.0, end 1.0
+  const qc::AodOperation deactivate(qc::OpType::AodDeactivate, {0, 1},
+                                    {qc::Dimension::Y}, {0.0}, {1.0});
+  EXPECT_EQ(deactivate.getNqubits(), 2);
+  EXPECT_EQ(deactivate.getStarts(qc::Dimension::Y).at(0), 0.0);
+  EXPECT_EQ(deactivate.getEnds(qc::Dimension::Y).at(0), 1.0);
+}
+
+TEST(AodOperation, Move) {
+  // move from 0,1 to 2,3 dimension X, start 0.0, end 1.0 and dimension Y,
+  // start 1.0, end 2.0
+  const qc::AodOperation move(qc::OpType::AodMove, {0, 1},
+                              {qc::Dimension::X, qc::Dimension::Y}, {0.0, 1.0},
+                              {1.0, 2.0});
+  EXPECT_EQ(move.getNqubits(), 2);
+  EXPECT_EQ(move.getStarts(qc::Dimension::X).at(0), 0.0);
+  EXPECT_EQ(move.getEnds(qc::Dimension::X).at(0), 1.0);
+  EXPECT_EQ(move.getStarts(qc::Dimension::Y).at(0), 1.0);
+  EXPECT_EQ(move.getEnds(qc::Dimension::Y).at(0), 2.0);
+}
+
+TEST(AodOperation, Distances) {
+  const qc::AodOperation move(qc::OpType::AodMove, {0, 1},
+                              {qc::Dimension::X, qc::Dimension::Y}, {0.0, 1.0},
+                              {1.0, 3.0});
+  EXPECT_EQ(move.getMaxDistance(qc::Dimension::X), 1.0);
+  EXPECT_EQ(move.getMaxDistance(qc::Dimension::Y), 2.0);
+}
+
+TEST(AodOperation, Qasm) {
+  const qc::AodOperation move(qc::OpType::AodMove, {0, 1},
+                              {qc::Dimension::X, qc::Dimension::Y}, {0.0, 1.0},
+                              {1.0, 3.0});
+  std::stringstream ss;
+  qc::RegisterNames qreg;
+  qreg.emplace_back("q", "q[0]");
+  qreg.emplace_back("q", "q[1]");
+  qc::RegisterNames const creg{{"c", "c"}};
+  move.dumpOpenQASM(ss, qreg, creg, 0, false);
+
+  EXPECT_FALSE(ss.str().empty());
```

Comment:
this doesn't really test anything. Couldn't you just test the actual string that results?
Or, potentially even better, create a circuit with the operation, dump it and compare against a reference.

Reply:
Added a test to compare to the actually correct string.